src/main/java/de/sakpaas/backend/service/SearchMappingService.java

```diff
@@ -0,0 +1,81 @@
+package de.sakpaas.backend.service;
+
+import com.google.common.annotations.VisibleForTesting;
+import de.sakpaas.backend.dto.NominatimSearchResultListDto;
+import de.sakpaas.backend.dto.NominatimSearchResultListDto.NominatimResultLocationDto;
+import de.sakpaas.backend.model.CoordinateDetails;
+import java.util.ArrayList;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.DefaultUriBuilderFactory;
+
+@Service
+public class SearchMappingService {
+
+  @Value("${app.search-api-url}")
+  private String searchApiUrl;
+
+  /**
+   * Searches in the Nominatim Microservice for the given key.
+   *
+   * @param key The search parameter. Multiple words are separated with %20.
+   * @return The list of Locations in our database
+   */
+  public CoordinateDetails search(String key) {
+    // Makes a request to the Nominatim Microservice
+    final String url = this.searchApiUrl + "/search/" + key + "?format=json";
+    final List<NominatimResultLocationDto> list = makeRequest(url);
+    return calculateCenter(list);
+  }
+
+  /**
+   * Makes an REST request to the provided URL.
+   *
+   * @param url The URL the request points to
+   * @return A List of NominatimResultLocationDto
+   */
+  @VisibleForTesting
+  protected List<NominatimResultLocationDto> makeRequest(String url) {
+    DefaultUriBuilderFactory defaultUriBuilderFactory = new DefaultUriBuilderFactory();
+    defaultUriBuilderFactory.setEncodingMode(DefaultUriBuilderFactory.EncodingMode.NONE);
+    RestTemplate restTemplate = new RestTemplate();
+    HttpHeaders httpHeaders = new HttpHeaders();
+    httpHeaders.set(HttpHeaders.ACCEPT, "text/html");
+    HttpEntity<String> entityReq = new HttpEntity<>(httpHeaders);
+    restTemplate.setUriTemplateHandler(defaultUriBuilderFactory);
+    ResponseEntity<NominatimSearchResultListDto> response =
+        restTemplate.exchange(url, HttpMethod.GET, entityReq, NominatimSearchResultListDto.class);
+    if (response.getBody() == null) {
+      return new ArrayList<>();
+    }
+
+    return response.getBody().getElements();
+  }
+
+  /**
+   * Calculates the central coordinates of a list of NominatimResultLocationDto.
+   *
+   * @param list The list of NominatimResultLocationDto
+   * @return The central coordinates
+   */
+  @VisibleForTesting
+  protected CoordinateDetails calculateCenter(List<NominatimResultLocationDto> list) {
+
+    double latitude = list.stream()
+        .map(NominatimResultLocationDto::getLat)
+        .mapToDouble(lat -> lat)
+        .average().orElse(0.0);
+    double longitude = list.stream()
```

Comment:
Hier kommen wenn man z.B. nach Berlin sucht sehr unsinnige Koordinaten raus, weil es in Amerika wohl auch ein Berlin gibt. Eventuell nur auf die erste Koordinate gehen?

![image](https://user-images.githubusercontent.com/35638696/79855827-c9796f80-83cb-11ea-9fe4-88d492750f3f.png)

Reply:
Und was ist, wenn du dann nach dem Berlin in Amerika suchst?

Reply:
Ich glaube Nominatim sortiert schon nach Relevanz. Daher sollte es schon passen, wenn man das erste Ergebniss nimmt.

Reply:
@RobSpot-Gaming `url = url.replace(" ", "%20");` das löst das Problem 👍